### ISSUE TITLE
Pax sgefault under ARM

### ISSUE THREAD

**Issue by alphaonex86** [google/protobuf#1161]
Hello:
under grsec/PaX:
[511438.529097] PAX: From 192.168.158.10: execution attempt in: (null), 00000000-00000000 00000000
[511438.530870] PAX: terminating task: /var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1-.arm/src/.libs/lt-protoc(lt-protoc):615, uid/euid: 250/250, PC: ffff0fc0, SP: b55b97b0
[511438.546024] PAX: bytes at PC: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 
[511438.546058] PAX: bytes at SP-4: 00000000 0000000c 9f3dbe00 b55b9844 9f808b28 9fbdbec0 9f80a0c0 b55b97e0 b55b988c 9fbdc908 9f72b3c8 b55b9844 9f72bfc0 9f807558 9f72a77c 00000000 9f3dbe00 b55b9844 9f8073d0 9f8073e8 00000011 
[511438.546134] grsec: From 192.168.158.10: denied resource overstep by requesting 4096 for RLIMIT_CORE against limit 0 for /var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1-.arm/src/.libs/lt-protoc[lt-protoc:615] uid/euid:250/250 gid/egid:250/250, parent /bin/bash[sh:614] uid/euid:250/250 gid/egid:250/250
libtool: link: armv7a-hardfloat-linux-gnueabi-g++ -pthread -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -O2 -pipe -march=native -mtune=native -mfpu=vfpv3 -mfloat-abi=hard -DGOOGLE_PROTOBUF_NO_RTTI -pthread -Wl,-O1 -o .libs/protoc google/protobuf/compiler/main.o  -Wl,--as-needed ./.libs/libprotobuf.so ./.libs/libprotoc.so /var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1-.arm/src/.libs/libprotobuf.so -lpthread -lz -pthread
oldpwd=`pwd` && ( cd /var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1/src && $oldpwd/protoc -I. --cpp_out=$oldpwd google/protobuf/unittest.proto google/protobuf/unittest_empty.proto google/protobuf/unittest_import.proto google/protobuf/unittest_import_public.proto google/protobuf/unittest_mset.proto google/protobuf/unittest_optimize_for.proto google/protobuf/unittest_embed_optimize_for.proto google/protobuf/unittest_custom_options.proto google/protobuf/unittest_lite.proto google/protobuf/unittest_import_lite.proto google/protobuf/unittest_import_public_lite.proto google/protobuf/unittest_lite_imports_nonlite.proto google/protobuf/unittest_no_generic_services.proto google/protobuf/compiler/cpp/cpp_test_bad_identifiers.proto )
/bin/sh: line 1: 17144 Killed                  $oldpwd/protoc -I. --cpp_out=$oldpwd google/protobuf/unittest.proto google/protobuf/unittest_empty.proto google/protobuf/unittest_import.proto google/protobuf/unittest_import_public.proto google/protobuf/unittest_mset.proto google/protobuf/unittest_optimize_for.proto google/protobuf/unittest_embed_optimize_for.proto google/protobuf/unittest_custom_options.proto google/protobuf/unittest_lite.proto google/protobuf/unittest_import_lite.proto google/protobuf/unittest_import_public_lite.proto google/protobuf/unittest_lite_imports_nonlite.proto google/protobuf/unittest_no_generic_services.proto google/protobuf/compiler/cpp/cpp_test_bad_identifiers.proto
Makefile:3647: recipe for target 'unittest_proto_middleman' failed
make[2]: *** [unittest_proto_middleman] Error 137
make[2]: Leaving directory '/var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1-.arm/src'
Makefile:710: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/dev-libs/protobuf-2.6.1-r3/work/protobuf-2.6.1-.arm'
Makefile:617: recipe for target 'all' failed
make: *** [all] Error 2
 * ERROR: dev-libs/protobuf-2.6.1-r3::gentoo failed (compile phase):

https://github.com/google/protobuf/tree/protoc-alpha-3.1:
Same.
But I need the 2.6.x stabilisation.

Cheers,

**Comment by alphaonex86** [comment]
Up, this is potential security problem!

**Comment by xfxyjwf** [comment]
@alphaonex86 Sorry, but I couldn't get much info from your posted error message. I don't have access to a computer with PaX either. Could you help debug and find out the cause of the problem?

**Comment by alphaonex86** [comment]
Yes, as C dev I can provide you some basic info, give the process and arg, and I will test with gdb.
I can in private too provide you a ssh access, normal user should have the same bug.
